helix-vcs/src/git.rs

```diff
@@ -1,17 +1,65 @@
-use std::path::Path;
+use std::{
+    collections::{hash_map::Entry, HashMap, HashSet},
+    io::Read,
+    path::{Path, PathBuf},
+};
 
+use anyhow::Result;
+use git::index::{entry::Mode, State};
 use git::objs::tree::EntryMode;
-use git::sec::trust::DefaultForLevel;
+use git::{prelude::FindExt, sec::trust::DefaultForLevel};
 use git::{Commit, ObjectId, Repository, ThreadSafeRepository};
 use git_repository as git;
+use ignore::WalkBuilder;
+use sha1::Digest;
 
-use crate::DiffProvider;
+use crate::{DiffProvider, FileChange};
 
 #[cfg(test)]
 mod test;
 
 pub struct Git;
 
+/// A subset of `git_repository::objs::tree::EntryMode` that actually makes sense for tree nodes.
```

Comment:
You can use `[..]` so that people building the doc locally will have a link to the enum

```suggestion
/// A subset of [`git_repository::objs::tree::EntryMode`] that actually makes sense for tree nodes.
```